Netkan/Validators/PluginsValidator.cs

```diff
@@ -0,0 +1,81 @@
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json.Linq;
+using ICSharpCode.SharpZipLib.Zip;
+using log4net;
+using CKAN.NetKAN.Services;
+using CKAN.NetKAN.Model;
+using CKAN.Games;
+
+namespace CKAN.NetKAN.Validators
+{
+    internal sealed class PluginsValidator : IValidator
+    {
+        public PluginsValidator(IHttpService http, IModuleService moduleService)
+        {
+            _http          = http;
+            _moduleService = moduleService;
+        }
+
+        public void Validate(Metadata metadata)
+        {
+            Log.Info("Validating that metadata is appropriate for DLLs");
+
+            JObject    json = metadata.Json();
+            CkanModule mod  = CkanModule.FromJson(json.ToString());
+            if (!mod.IsDLC)
+            {
+                var package = _http.DownloadModule(metadata);
+                if (!string.IsNullOrEmpty(package))
+                {
+                    ZipFile      zip  = new ZipFile(package);
+                    GameInstance inst = new GameInstance(new KerbalSpaceProgram(), "/", "dummy", new NullUser());
+
+                    var plugins    = _moduleService.GetPlugins(mod, zip, inst).ToList();
+                    bool hasPlugin = plugins.Any();
+                    if (hasPlugin)
+                    {
+                        var dllPaths = plugins.Select(f => inst.ToRelativeGameDir(f.destination)).ToList();
+                        var pattern = Registry.DllPattern(inst.game);
+                        var dllIdentifiers = dllPaths
+                            .Select(p => pattern.Match(p))
+                            .Where(m => m.Success)
+                            .Select(m => m.Groups["modname"].Value)
+                            .Where(ident => ident != metadata.Identifier
```

Comment:
Hmm, this looks wrong. Will need to figure out what I was going for here and reconsider, but I can't do that right now.

Reply:
Well whatever it was, I'm pretty sure it was wrong, so I took it out. Hopefully if there was a valid idea here, we'll rediscover it some other way.